pmultiqc/modules/quantms/quantms.py

```diff
@@ -1032,42 +912,19 @@ def dis_decoy(ProteinName):
                 return "DECOY"
 
     def parse_mzml(self):
-        if self.enable_dia:
-            self.peak_intensity_distribution['Whole Experiment'] = dict(zip(
-                ['>10000', '6000-10000', '3000-6000', '1000-3000', '900-1000', '700-900',
-                '500-700', '300-500', '100-300', '10-100', '0-10'], [0] * 11
-            ))
-            self.charge_state_distribution['Whole Experiment'] = dict(zip(
-                ['>7', '7', '6', '5', '4', '3', '2', '1'], [0] * 8
-            ))
-            self.peak_per_ms2['Whole Experiment'] = dict(zip(
-                ['>1000', '900-1000', '800-900', '700-800', '600-700', '500-600', '400-500',
-                '300-400', '200-300', '100-200', '0-100'], [0] * 11
-            ))
-        else:
-            self.peak_intensity_distribution['identified_spectra'] = dict(zip(
-                ['>10000', '6000-10000', '3000-6000', '1000-3000', '900-1000', '700-900',
-                '500-700', '300-500', '100-300', '10-100', '0-10'], [0] * 11
-            ))
-            self.peak_intensity_distribution['unidentified_spectra'] = dict(zip(
-                ['>10000', '6000-10000', '3000-6000', '1000-3000', '900-1000', '700-900',
-                '500-700', '300-500', '100-300', '10-100', '0-10'], [0] * 11
-            ))
-            self.charge_state_distribution['identified_spectra'] = dict(zip(
-                ['>7', '7', '6', '5', '4', '3', '2', '1'], [0] * 8
-            ))
-            self.charge_state_distribution['unidentified_spectra'] = dict(zip(
-                ['>7', '7', '6', '5', '4', '3', '2', '1'], [0] * 8
-            ))
-            self.peak_per_ms2['identified_spectra'] = dict(zip(
-                ['>1000', '900-1000', '800-900', '700-800', '600-700', '500-600', '400-500',
-                '300-400', '200-300', '100-200', '0-100'], [0] * 11
-            ))
-
-            self.peak_per_ms2['unidentified_spectra'] = dict(zip(
-                ['>1000', '900-1000', '800-900', '700-800', '600-700', '500-600', '400-500',
-                '300-400', '200-300', '100-200', '0-100'], [0] * 11
-            ))
+        data_cats_list = ['Whole Experiment'] if self.enable_dia else ['identified_spectra', 'unidentified_spectra']
```

Comment:
I think I would leave that out of the histogram class. Can't you create a nested dict?

E.g. in the end for plotting create:
```
dict ( "Whole Experiment": histogram.to_dict) ?